packages/next/build/webpack/loaders/next-serverless-loader.ts

```diff
@@ -80,12 +80,20 @@ const nextServerlessLoader: loader.Loader = function () {
           .reduce((prev, key) => {
             let value = query[key]
 
+            ${
+              ''
+              // non-provided optional values should be undefined so normalize
+              // them to undefined
+            }
+            if(routeRegex.groups[key].optional && !value) {
+              value = undefined
```

Comment:
We should skip setting the key below if this undefined makes it all the way through, right? Otherwise it'd be in the `Object.keys`.

Reply:
Yeah, updated to make sure the key isn't set when it's missing. To do this I removed the un-necessary re-parsing of the query in the `apiResolver` it seems it was re-parsing and providing the same values we were providing in the params argument